packages/layout/src/Legend.ts

```diff
@@ -222,20 +222,42 @@ export class Legend extends SVGWidget {
         const ordinal = d3ScaleOrdinal()
             .domain(dataArr.map(row => row[1]))
             .range(dataArr.map(row => row[0]));
-
+        let total = 0;
         this._legendOrdinal
             .orient(this.orientation())
             .title(this.title())
             .scale(ordinal)
+            .labels(d => {
+                if (this.showSeriesTotal()) {
+                    const val = this.data()[d.i].slice(1).reduce((acc, n) => acc + n, 0);
+                    total += val;
+                    return `${d.domain[d.i]} (${val})`;
+                }
+                return d.domain[d.i];
+            })
             ;
         this._g.call(this._legendOrdinal);
 
         this.updateDisabled(element, dataArr);
 
         const bbox = this.getBBox(true, true);
         this._g.attr("transform", `translate(${this.width() / 2 - bbox.width / 2 + 5},${this.height() / 2 - bbox.height / 2})`);
+        const legendCellsBbox = this._g.select(".legendCells").node().getBBox();
+        const legendCellHeight = legendCellsBbox.height / (dataArr.length || 1);
+        const legendTotal = this._g.selectAll(".legendTotal").data(this.showLegendTotal() ? [total] : []);
+        const firstLabel = this._g.select(".label");
+        if (firstLabel.node()) {
```

Comment:
Should be `!firstLabel.empty()` (I think).

Reply:
If you moved lines 230 -> 237 into a local variable "legendLabels" then you can use "legendLabels.length" here to test if there are any labels and then change your selection logic to:
```
this._g.selectAll(".legendTotal").data(this.showLegendTotal() && legendLabels.length ? [total] : []);
```

Then you won't need any of this "checking" to see if the legend exists AND it will clean itself up when the data goes from 10 rows to 0 rows...